Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Confirm.hs

```diff
@@ -84,9 +84,14 @@ confirm transporterId (SignatureAuthResult _ subscriber) reqV2 = withFlowHandler
               fork "on_confirm/on_update" $ do
                 handle (errHandler dConfirmRes transporter (Just rideInfo'.driver)) $ do
                   -- We assign driver post `on_confirm`, so removing `rideInfo` details fron `dConfirmRes`.
-                  let dConfirmRes' = dConfirmRes {DConfirm.rideInfo = Nothing}
-                  callOnConfirm dConfirmRes' msgId txnId bapId callbackUrl bppId bppUri city country
-                  void $ BP.sendRideAssignedUpdateToBAP dConfirmRes.booking rideInfo'.ride rideInfo'.driver rideInfo'.vehicle
+                  -- let dConfirmRes' = dConfirmRes {DConfirm.rideInfo = Nothing}
+                  -- callOnConfirm dConfirmRes msgId txnId bapId callbackUrl bppId bppUri city country
+                  -- void $ BP.sendRideAssignedUpdateToBAP dConfirmRes.booking rideInfo'.ride rideInfo'.driver rideInfo'.vehicle
+                  context' <- ContextV2.buildContextV2 Context.CONFIRM Context.MOBILITY msgId txnId bapId callbackUrl bppId bppUri city country (Just "PT2M")
+                  let vehicleCategory = Utils.mapServiceTierToCategory dConfirmRes.booking.vehicleServiceTier
+                  becknConfig <- QBC.findByMerchantIdDomainAndVehicle dConfirmRes.transporter.id (show Context.MOBILITY) vehicleCategory >>= fromMaybeM (InternalError "Beckn Config not found")
+                  onConfirmMessage <- BP.buildOnConfirmMessage dConfirmRes.booking rideInfo'.ride rideInfo'.driver rideInfo'.vehicle
```

Comment:
make sendOnConfirmToBAP, and put buildOnConfirmMessage inside this function itself.